### PR TITLE
Source Azure Blob Storage: support  OAuth

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve https://github.com/airbytehq/airbyte-internal-issues/issues/7049

## How

1. add support for oauth2.0
2. add `config_migration` to align with new config format (move `azure_blob_storage_account_key` inside `credentials` object)

### Linked PR for cloud 
https://github.com/airbytehq/airbyte-platform-internal/pull/11936

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]
lgtm

File: airbyte-integrations/connectors/source-azure-blob-storage/source_azure_blob_storage/config.py [hunk 1]
```diff
@@ -2,11 +2,51 @@
...
+from typing import Any, Dict, Literal, Optional, Union
 
 import dpath.util
 from airbyte_cdk.sources.file_based.config.abstract_file_based_spec import AbstractFileBasedSpec
-from pydantic import AnyUrl, Field
+from airbyte_cdk.utils.oneof_option_config import OneOfOptionConfig
+from pydantic import AnyUrl, BaseModel, Field
+
+
+class Oauth2(BaseModel):
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-azure-blob-storage/source_azure_blob_storage/config.py:13]: looks like breaking changes

Comment (artem1205) [reply airbyte-integrations/connectors/source-azure-blob-storage/source_azure_blob_storage/config.py:13]: config_migration script will prevent breaking changes

File: airbyte-integrations/connectors/source-azure-blob-storage/integration_tests/configs/config_integration_csv.json
Comment (davidvaldez89d) [inline airbyte-integrations/connectors/source-azure-blob-storage/integration_tests/configs/config_integration_csv.json:1]: hi, this change broke my terraform code, should I modify my resource block to something like this?
 configuration = {
    credentials = {
    auth_type = "storage_account_key"
    azure_blob_storage_account_key    = azurerm_storage_account.az_st.primary_access_key
    }
    azure_blob_storage_account_name   = azurerm_storage_account.az_st.name
    azure_blob_storage_container_name = azurerm_storage_container.az_cn.name
    azure_blob_storage_endpoint       = azurerm_storage_account.az_st.primary_blob_endpoint
    streams = {...}
    }
 configuration = {
    azure_blob_storage_account_name   = azurerm_storage_account.az_st.name
    azure_blob_storage_container_name = azurerm_storage_container.az_cn.name
    azure_blob_storage_account_key    = azurerm_storage_account.az_st.primary_access_key
    azure_blob_storage_endpoint       = azurerm_storage_account.az_st.primary_blob_endpoint
    streams = {...}
    }

**Pull request merged**